include/picongpu/plugins/output/images/Visualisation.hpp

```diff
@@ -157,286 +159,654 @@ struct typicalFields < 5 >
-        //                                   visPreview::preChannel3_opacity);
-
-
-        // draw to (perhaps smaller) image cell
-        image(imageCell) = pic;
+        DataSpace< simDim > const suplercellIdx = mapper.getSuperCellIndex( DataSpace< simDim >( blockIdx ) );
+        // offset of the supercell (in cells) to the origin of the local domain
+        DataSpace< simDim > const supercellCellOffset(
+            ( suplercellIdx - mapper.getGuardingSuperCells( ) ) * SuperCellSize::toRT( )
+        );
+
+        using SupercellDomCfg = IdxConfig<
+            cellsPerSupercell,
+            numWorkers
+        >;
+
+        // each cell in a supercell is handled as a virtual worker
+        ForEachIdx< SupercellDomCfg > forEachCell( workerIdx );
+
+        forEachCell(
+            [&](
+                uint32_t const linearIdx,
+                uint32_t const
+            )
+            {
+                // cell index within the superCell
+                DataSpace< simDim > const cellIdx = DataSpaceOperations< simDim >::template map< SuperCellSize >( linearIdx );
+                // offset to the origin of the local domain + guarding cells
+                DataSpace< simDim > const cellOffset( suplercellIdx * SuperCellSize::toRT() + cellIdx );
+                // cell offset without guarding cells
+                DataSpace< simDim > const realCell( supercellCellOffset + cellIdx );
+                // offset within the two dimensional result buffer
+                DataSpace< DIM2 > const imageCell(
+                    realCell[ transpose.x( ) ],
+                    realCell[ transpose.y( ) ]
+                );
+
+                bool const isCellOnSlice = IsPartOfSlice< >{}(
+                    realCell,
+                    sliceDim,
+                    localDomainOffset,
+                    slice
+                );
+
+                /* if the virtual worker is not calculating a cell out of the
+                 * selected slice than exit
+                 */
+                if( !isCellOnSlice )
+                    return;
+
+                // set fields of this cell to vars
+                typename T_BBox::ValueType field_b = fieldB( cellOffset );
+                typename T_EBox::ValueType field_e = fieldE( cellOffset );
+                typename T_JBox::ValueType field_j = fieldJ( cellOffset );
+
+                // multiply with the area size of each plane
+                field_j *= float3_X::create( CELL_VOLUME ) / cellSize;
+
+                /* reset picture to black
+                 *   color range for each RGB channel: [0.0, 1.0]
+                 */
+                float3_X pic(
+                    // typical values of the fields to normalize them to [0,1]
+                    visPreview::preChannel1(
+                        field_b / typicalFields< EM_FIELD_SCALE_CHANNEL1 >::get( ).x( ),
```

Comment:
OPTIONAL:
using `.x()   .y()   .z()` in this context is highly confusing (for a physicist) since it looks like you access the component of the vector field not the (x=B, y=E, z=J) normalization of the visualization plugin. 
Might be less confusing with array index numbers (if possible). 
*definitely not a required change*

Reply:
@psychocoderHPC 🎉 Thank you for changing this!